### ISSUE TITLE
Tenant connection is set to null after tenant model is updated in the database

### ISSUE THREAD

**Issue by aislanfoina** [tenancy/tenancy#239]
__Description__
Tenant connection is set to null after tenant model is updated in the database

---

__Actual behavior__
When some info in the current tenant needs to be updated, after the row is updated in the DB, some internal hooks (I think it is the DatabaseMutation Hook) are called and the  tenant connection is set to null. This cause laravel to throw error if any query is executed in the tenant connection inside the same call.

__Expected behavior__
Tenant connection should remain valid, otherwise most of the time when the user updates any info on tenant model laravel with throw error, unless the developer avoid any db query after the update.

---
__Information__

- tenancy/tenancy version:  1.3.0
- laravel version: 8.68.1
- database driver and version: Mysql
- webserver software and version: Apache
- php version: 7.4

---

**Comment by thumbtech** [comment]
I use the service extensively (typically http identification) and have not had this issue.

Do you see the problem right out of the box (base laravel +tenancy) ... ?

**Comment by sniper7kills** [comment]
@thumbtech I believe its *only* when the queue is set to "sync" so any of the hooks run in the browser request instead of in a queue outside of the request; but that's simply based on my limited memory/knowledge of the discord conversation about it.

**Comment by sniper7kills** [comment]
@ArlonAntonius I believe this is related to #233 or I had looked into a similar issue few months back.

**Comment by aislanfoina** [comment]
@thumbtech yes, I'm not doing anything special/wierd. It is just an update on mysql tenant row. I was able to walk around creating another table and do a 1-to-1 relationship and apply the update to this table. 

Did you try to run an update on your tenant model on the same tenant you are using, on a multi database tenant deployment?

**Comment by thumbtech** [comment]
Oh I see this is a different use case since I am using model affect (1 db), NOT connection. Sorry for confusion!

**Comment by ArlonAntonius** [comment]
This is an issue in the MySQL driver as far as I have seen. The MySQL driver creates a new DB connection on the tenant connection name that is causing the connection to also empty. Simply adjusting the logic related to the connection creation would fix the specified issue.

https://github.com/tenancy/tenancy/blob/f97fe4efe951151377d72b340461ae2d34c1c32b/src/Database/Mysql/Driver/Mysql.php#L134-L147

**Comment by luceos** [comment]
@ArlonAntonius does this still require a change to the driver?

**Comment by ArlonAntonius** [comment]
Yes, should use a different connection name
